### PR TITLE
Added json content type for apns, gcm, template notifications to correspond to api spec.

### PR DESCRIPTION
**Pull request by olgaakv**

### CODE REVIEW

File: src/Microsoft.Azure.NotificationHubs/AdmNotification.cs [hunk 1]
```diff
@@ -13,12 +13,14 @@ namespace Microsoft.Azure.NotificationHubs
     /// </summary>
     public sealed class AdmNotification : Notification, INativeNotification
     {
+        const string contentType = "application/xml";
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs/AdmNotification.cs:16]: ADM uses the same as Firebase and GCM, which is JSON, not XML

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/NotificationContentTypeTest.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        [Fact]
+        public void NotificationContentTypeCorrespondsToApi()
+        {
+            Assert.Equal("application/json", new AppleNotification("{\"aps\":{\"alert\":\"alert!\"}}").ContentType);
+            Assert.Equal("application/json", new FcmNotification("{\"data\":{\"message\":\"Message\"}}").ContentType);
+            Assert.Equal("application/xml", new MpnsNotification("<wp:Notification xmlns:wp=\"WPNotification\" Version=\"2.0\"><wp:Tile Id=\"TileId\" Template=\"IconicTile\"><wp:Title Action=\"Clear\">Title</wp:Title></wp:Tile></wp:Notification>").ContentType);
+            Assert.Equal("application/xml", new WindowsNotification("<toast><visual><binding template=\"ToastText01\"><text id=\"1\">bodyText</text></binding>  </visual></toast>").ContentType);
+            Assert.Equal("application/json", new TemplateNotification(new Dictionary<string, string>()).ContentType);
+            Assert.Equal("application/xml", new AdmNotification("{\"data\":{\"key1\":\"value1\"}}").ContentType);
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/NotificationContentTypeTest.cs:16]: ADM is not XML, as noted in the [ADM Documentation](https://developer.amazon.com/docs/adm/send-message.html)

File: src/Microsoft.Azure.NotificationHubs/MpnsNotification.cs [hunk 3]
```diff
@@ -15,6 +15,8 @@ namespace Microsoft.Azure.NotificationHubs
     /// </summary>
     public sealed class MpnsNotification : Notification, INativeNotification
     {
+        const string contentType = "application/xml";
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs/MpnsNotification.cs:18]: As noted in [our own documentation](https://docs.microsoft.com/en-us/rest/api/notificationhubs/send-mpns-native-notification) it should be `application/xml;charset=utf-8`

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/NotificationContentTypeTest.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+namespace Microsoft.Azure.NotificationHubs.DotNetCore.Tests
+{
+    public class NotificationContentTypeTest
+    {
+        [Fact]
+        public void NotificationContentTypeCorrespondsToApi()
+        {
+            Assert.Equal("application/json", new AppleNotification("{\"aps\":{\"alert\":\"alert!\"}}").ContentType);
+            Assert.Equal("application/json", new FcmNotification("{\"data\":{\"message\":\"Message\"}}").ContentType);
+            Assert.Equal("application/xml", new MpnsNotification("<wp:Notification xmlns:wp=\"WPNotification\" Version=\"2.0\"><wp:Tile Id=\"TileId\" Template=\"IconicTile\"><wp:Title Action=\"Clear\">Title</wp:Title></wp:Tile></wp:Notification>").ContentType);
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/NotificationContentTypeTest.cs:13]: As noted in ]our own documentation](https://docs.microsoft.com/en-us/rest/api/notificationhubs/send-mpns-native-notification) it should be application/xml;charset=utf-8

File: src/Microsoft.Azure.NotificationHubs/WindowsNotification.cs [hunk 5]
```diff
@@ -15,6 +15,7 @@ namespace Microsoft.Azure.NotificationHubs
     /// </summary>
     public sealed class WindowsNotification : Notification, INativeNotification
     {
+        const string contentType = "application/xml";
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs/WindowsNotification.cs:18]: The content-type for WNS is variable as noted in [our own documentation](https://docs.microsoft.com/en-us/rest/api/notificationhubs/send-wns-native-notification)

Set to `application/json;charset=utf-8` or `application/xml`. If the notification type (`X-WNS-Type`) is `wns/raw`, set to `application/octet-stream`.

File: src/Microsoft.Azure.NotificationHubs/TemplateNotification.cs [hunk 6]
```diff
@@ -15,6 +15,7 @@ namespace Microsoft.Azure.NotificationHubs
     /// </summary>
     public sealed class TemplateNotification : Notification
     {
+        const string contentType = "application/json";
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs/TemplateNotification.cs:18]: Templates should be `application/json;charset=utf-8` according to [our own documentation](https://docs.microsoft.com/en-us/rest/api/notificationhubs/send-template-notification)

File: src/Microsoft.Azure.NotificationHubs/WindowsNotification.cs (hunk 5 above)
Comment (olgaakv) [reply src/Microsoft.Azure.NotificationHubs/WindowsNotification.cs:18]: Currently it is set up [here ](https://github.com/Azure/azure-notificationhubs-dotnet/blob/d203f2b0c6cafb07b8f1e799c6983423f34c811e/src/Microsoft.Azure.NotificationHubs/WindowsNotification.cs#L134)

**Review by mpodwysocki — APPROVED** [review]

**Pull request merged**